### ISSUE TITLE
mysql is not booting: No space left on device

### ISSUE THREAD

**Issue by nadworny** [kubevious/kubevious#50]
### Describe the bug
After deploying using helm chart the mysql is in a crashloopbackoff state all the time. Here is the error log:
```
mysql 2020-12-02 20:43:15+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.22-1debian10 started.                                                         
mysql 2020-12-02 20:43:16+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'                                                                                  
mysql 2020-12-02 20:43:16+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.22-1debian10 started.                                                         
mysql 2020-12-02T20:43:16.346666Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.22) starting as process 1                                                  
mysql 2020-12-02T20:43:16.355244Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.                                                                      
mysql 2020-12-02T20:43:17.534375Z 1 [Warning] [MY-012637] [InnoDB] 1048576 bytes should have been written. Only 720896 bytes written. Retrying for the remaining bytes.   
mysql 2020-12-02T20:43:17.534530Z 1 [Warning] [MY-012638] [InnoDB] Retry attempts for writing partial data failed.                                                        
mysql 2020-12-02T20:43:17.534645Z 1 [ERROR] [MY-012639] [InnoDB] Write to file ./ibtmp1 failed at offset 3145728, 1048576 bytes should have been written, only 720896 were
mysql 2020-12-02T20:43:17.534766Z 1 [ERROR] [MY-012640] [InnoDB] Error number 28 means 'No space left on device'                                                          
mysql 2020-12-02T20:43:17.534993Z 1 [ERROR] [MY-012267] [InnoDB] Could not set the file size of './ibtmp1'. Probably out of disk space                                    
mysql 2020-12-02T20:43:17.535110Z 1 [ERROR] [MY-012926] [InnoDB] Unable to create the shared innodb_temporary.                                                            
mysql 2020-12-02T20:43:17.535234Z 1 [ERROR] [MY-012930] [InnoDB] Plugin initialization aborted with error Generic error.                                                  
mysql 2020-12-02T20:43:17.922041Z 1 [ERROR] [MY-010334] [Server] Failed to initialize DD Storage Engine                                                                   
mysql 2020-12-02T20:43:17.922357Z 0 [ERROR] [MY-010020] [Server] Data Dictionary initialization failed.                                                                   
mysql 2020-12-02T20:43:17.922691Z 0 [ERROR] [MY-010119] [Server] Aborting                                                                                                 
mysql 2020-12-02T20:43:17.923157Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.22)  MySQL Community Server - GPL.                      
```

### To Reproduce
Install using helm chart 

### Expected behavior
mysql not crashing.

### Environment Details:
 - aks v1.16.13
 - Version v0.7.26

**Comment by rubenhak** [comment]
@nadworny, there are two reasons why this might happen. How long had you been running Kubevious? How large is your deployment? How many nodes, pods, services, configmaps, etc do you have?

## Default size just not enough
It could be your deployment is larger then anticipated and 20Gb default storage size is just not enough. Unfortunately helm wouldn't let you update the "mysql.persistence.size" variable after the chart is already installed. You would also increase the size of the PVC **data-kubevious-mysql-0** and the corresponding PV. If you decide to increase the storage size, don't forget to restart the **mysql** pod to. Sometimes

## MySQL binary logs taking space
Even though Kubevious history db does auto-cleanup (after 2 weeks), there are still MySQL binary logs left behind. We are working on making this automatic. Meanwhile, you would follow this procedure to do the cleanup manually. Would probably take 1-2 minutes:

SSH into the mysql pod:
```sh
$ kubectl exec -it kubevious-mysql-0 -n kubevious sh
```

Inside the ssh session check how much memory is taken by the index files:
```sh
$ ls -la /var/lib/mysql
```

You would probably see bunch of files like this:
```
-rw-r----- 1 mysql mysql  602523276 Nov 25 07:25  kubevious-mysql-0-bin.000013
-rw-r----- 1 mysql mysql   35220750 Nov 25 08:00  kubevious-mysql-0-bin.000014
-rw-r----- 1 mysql mysql   26848444 Nov 25 08:32  kubevious-mysql-0-bin.000015
-rw-r----- 1 mysql mysql 1073791064 Nov 26 06:27  kubevious-mysql-0-bin.000016
-rw-r----- 1 mysql mysql 1073774637 Nov 27 04:41  kubevious-mysql-0-bin.000017
-rw-r----- 1 mysql mysql 1074238073 Nov 28 03:04  kubevious-mysql-0-bin.000018
-rw-r----- 1 mysql mysql 1074723342 Nov 29 01:38  kubevious-mysql-0-bin.000019
-rw-r----- 1 mysql mysql 1073787584 Nov 30 00:13  kubevious-mysql-0-bin.000020
-rw-r----- 1 mysql mysql 1073749219 Nov 30 22:56  kubevious-mysql-0-bin.000021
-rw-r----- 1 mysql mysql 1073772048 Dec  1 21:53  kubevious-mysql-0-bin.000022
-rw-r----- 1 mysql mysql 1073756265 Dec  2 19:54  kubevious-mysql-0-bin.000023
-rw-r----- 1 mysql mysql  362574108 Dec  3 03:15  kubevious-mysql-0-bin.000024
-rw-r----- 1 mysql mysql        372 Dec  2 19:54  kubevious-mysql-0-bin.index
```

Execute the following command to purge logs. It will ask for root password. Adjust the date accordingly. 
```sh
$ mysql -u root -p -e "PURGE BINARY LOGS BEFORE '2020-12-03'"
```

You can get the root password by from the secret below. Remember that secret returns base64 encoded data values. Of course, run outside of the SSH shell.
```sh
$ kubectl get secret kubevious-mysql-secret-root -n kubevious -o yaml
```

**Comment by nadworny** [comment]
hi @rubenhak and thanks for a detailed explanation. I'll try it as soon as possible. I thought that by removing the helm chart which was the first thing I trued also the PVC should be cleaned up but I guess that's not the case?

**Comment by rubenhak** [comment]
> hi @rubenhak and thanks for a detailed explanation. I'll try it as soon as possible.
> 
> On the other side, I thought that by removing the helm chart which was the first thing I trued also the PVC should be cleaned up but I guess that's not the case?

Sure, n.p. Please get back once you get this running.

Unfortunately not. The PV would stay and would be remounted whenever the chart is reinstalled. There would be another issue with MySQL password if reinstalled. Please see the explanation and solution here: https://github.com/kubevious/helm#uninstalling-the-chart

**Comment by kubevious** [comment]
Fixed in [v0.8](https://github.com/kubevious/kubevious/releases/tag/v0.8)
